src/test/java/org/alfresco/rest/api/tests/TestActions.java

```diff
@@ -504,4 +510,97 @@ private void checkSorting(
+
+    @Test
+    public void testExecuteAction() throws Exception
+    {
+        final String person1 = account1PersonIt.next();
+        publicApiClient.setRequestContext(new RequestContext(account1.getId(), person1));
+
+        AuthenticationUtil.setFullyAuthenticatedUser(person1);
+
+        String myNode = getMyNodeId();
+        NodeRef validNode = nodeService.createNode(new NodeRef(StoreRef.STORE_REF_WORKSPACE_SPACESSTORE, myNode), ContentModel.ASSOC_CONTAINS,
+                QName.createQName("test", "test-ea-node"), ContentModel.TYPE_CONTENT).getChildRef();
+
+        // actionDefinitionId missing but required by the action.
+        {
+            actions.executeAction(new Action(), emptyParams, 400);
+        }
+
+        // Non-existent actionDefinitionId
+        {
+            Action action = new Action();
+            action.setActionDefinitionId("nonExistentActionDefId");
+
+            actions.executeAction(action, emptyParams, 404);
+        }
+
+        // targetId missing but required by the action.
+        {
+            Action action = new Action();
+            action.setActionDefinitionId(AddFeaturesActionExecuter.NAME);
+
+            actions.executeAction(action, emptyParams, 400);
+        }
+
+        // Non-existent targetId
+        {
+            NodeRef nodeRef = new NodeRef(StoreRef.STORE_REF_WORKSPACE_SPACESSTORE, "750a2867-ecfa-478c-8343-fa0e39d27be3");
+            assertFalse("Test pre-requisite: node must not exist", nodeService.exists(nodeRef));
+
+            Action action = new Action();
+            action.setActionDefinitionId(AddFeaturesActionExecuter.NAME);
+            action.setTargetId(nodeRef.getId());
+
+            actions.executeAction(action, emptyParams, 404);
+        }
+
+        // Missing mandatory params - action not executed.
+        {
+            Action action = new Action();
+            action.setActionDefinitionId(AddFeaturesActionExecuter.NAME);
+            action.setTargetId(validNode.getId());
+
+            actions.executeAction(action, emptyParams, 202);
+
+            Thread.sleep(1000);
+
+            assertFalse("Aspect versionable wasn't expected !", nodeService.hasAspect(validNode, ContentModel.ASPECT_VERSIONABLE));
+
+        }
+
+        // Check add versionable aspect.
+        {
+            assertFalse(nodeService.hasAspect(validNode, ContentModel.ASPECT_VERSIONABLE));
+
+            Map<String, String> params = new HashMap<>();
+            params.put(AddFeaturesActionExecuter.PARAM_ASPECT_NAME, "cm:versionable");
+
+            Action action = new Action();
+            action.setActionDefinitionId(AddFeaturesActionExecuter.NAME);
+            action.setTargetId(validNode.getId());
+            action.setParams(params);
+
+            Action executedAction = actions.executeAction(action, emptyParams, 202);
+            assertNotNull(executedAction);
+            assertNotNull(executedAction.getId());
+
+            new RetryingCallbackHelper.RetryingCallback()
```

Comment:
This won't get executed - it's just creating an instance that is then thrown away

Reply:
This part just checks that the returned result contains the generated id, a prerequisite for further methods that will be available in the future and will use this.

Reply:
No, the `RetryingCallbackHelper.RetryingCallback` is never executed, as you're creating the instance and throwing it away — you need to actually execute it by passing it to `RetryingCallbackHelper.doWithRetry(RetryingCallback)` for it do do anything.

Reply:
Will change that, thanks.